common/src/main/java/com/wynntils/gui/render/FontRenderer.java

```diff
@@ -82,6 +87,67 @@ public void renderText(
         }
     }
 
+    private void renderRainbowText(
+            PoseStack poseStack,
+            String text,
+            float x,
+            float y,
+            HorizontalAlignment horizontalAlignment,
+            VerticalAlignment verticalAlignment,
+            TextShadow shadow) {
+        float renderX;
+        float renderY;
+
+        renderX = switch (horizontalAlignment) {
+            case Left -> x;
+            case Center -> x - (font.width(text) / 2f);
+            case Right -> x - font.width(text);};
+
+        renderY = switch (verticalAlignment) {
+            case Top -> y;
+            case Middle -> y - (font.lineHeight / 2f);
+            case Bottom -> y - font.lineHeight;};
+
+        for (char c : text.toCharArray()) {
+            long dif = ((long) renderX * 10) - ((long) renderX * 10);
+
+            String s = String.valueOf(c);
+
+            // color settings
+            long time = System.currentTimeMillis() - dif;
+            float z = 2000.0F;
+
+            int colorInt = Color.HSBtoRGB((float) (time % (int) z) / z, 0.8F, 0.8F);
+            CustomColor color = CustomColor.fromInt(colorInt);
+
+            // rendering shadows
+            float originPosX = x;
+            float originPosY = y;
+            switch (shadow) {
```

Comment:
I am pretty sure you can call renderText (the original one) here so we don't need to duplicate this either

Reply:
I think it needs to be duplicated to correct for the alignment